### PR TITLE
Add Matplotlib label lines

### PR DESCRIPTION
**Pull request by kalling**
Pip package only has whl, so adapted from repo (https://github.com/cphyc/matplotlib-label-lines) releases.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/matplotlib-label-lines/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+
+  run:
+    - python >=3.6
+    - {{ pin_compatible('numpy') }}
```
Comment (dopplershift) [inline recipes/matplotlib-label-lines/meta.yaml:24]: You only need the `pin_compatible` if you're building against Numpy's C-API:
```suggestion
    - numpy
```

File: recipes/matplotlib-label-lines/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - matplotlib
```
Comment (dopplershift) [inline recipes/matplotlib-label-lines/meta.yaml:25]: Unless you need a PyQt-based GUI, better to depend on `matplotlib-base`:
```suggestion
    - matplotlib-base
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a couple minor tweaks to dependencies and this will be good to go. @pigattol can you confirm that you are willing to be a maintainer for this recipe?

File: recipes/matplotlib-label-lines/meta.yaml (hunk 2 above)
Comment (pigattol) [reply recipes/matplotlib-label-lines/meta.yaml:25]: 👍 @dopplershift I confirm

**Comment by orso82** [comment]
Thank you @kalling and @pigattol

**Review by dopplershift — APPROVED** [review]

**Pull request merged**